internal/lightcone/nihility/goodnightandsleepwell/goodnightandsleepwell.go

```diff
@@ -0,0 +1,56 @@
+package goodnightandsleepwell
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Check key.Modifier = "good_night_and_sleep_well_check"
+)
+
+func init() {
+	lightcone.Register(key.GoodNightandSleepWell, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_NIHILITY,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(Check, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHitAll: onBeforeHitAll,
+		},
+	})
```

Comment:
Add `CanModifySnapshot: true` to this config like how it is done in Fermata. This is how we mark modifiers as being usable in snapshots so it can apply to dots (our equivalent of the `OnSnapshotCreate` callback).